src/pykiso/test_result/multi_result.py

```diff
@@ -99,9 +116,46 @@ def startTest(self, test: Union[BasicTest, BaseTestSuite]) -> None:
 
         :param test: running testcase
         """
+        self.handle_log_file_strategy(test)
+
         for result in self.result_classes:
             result.startTest(test)
 
+    def handle_log_file_strategy(self, test: Union[BasicTest, BaseTestSuite]) -> None:
+        """Handle the log file strategy for the given test.
+
+        :param test: The test case or test suite being executed.
+        """
+        if self._log_file_strategy is None:
+            return
+
+        log_options = get_logging_options()
+
+        if test.__class__ in self._list_test_results and self._log_file_strategy == "testCase":
+            log_path = self.current_log_file
+        else:
+            log_file_name = util.strclass(test.__class__).replace(".", "_").replace("-", "_")
+            if self._log_file_strategy == "testRun":
+                log_file_name += "_" + test._testMethodName
+
+            log_file_name += f"_{datetime.today().strftime('%Y%d%m%H%M%S')}.log"
```

Comment:
Why are you not reusing `self.current_log_file` as base for the split?

Reply:
It has been changed to use the self.current_log_file.